### Importing Libraries

In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### Reading Data

Source: https://www.kaggle.com/gyani95/380000-lyrics-from-metrolyrics

In [6]:
lyrics = pd.read_csv('Data/380000-lyrics-from-metrolyrics/lyrics.csv',index_col='index')

### A Quick Look At Data

In [7]:
lyrics.head()

song  year           artist genre  \
index                                                 
0            ego-remix  2009  beyonce-knowles   Pop   
1         then-tell-me  2009  beyonce-knowles   Pop   
2              honesty  2009  beyonce-knowles   Pop   
3      you-are-my-rock  2009  beyonce-knowles   Pop   
4        black-culture  2009  beyonce-knowles   Pop   

                                                  lyrics  
index                                                     
0      Oh baby, how you doing?\nYou know I'm gonna cu...  
1      playin' everything so easy,\nit's like you see...  
2      If you search\nFor tenderness\nIt isn't hard t...  
3      Oh oh oh I, oh oh oh I\n[Verse 1:]\nIf I wrote...  
4      Party the people, the people the party it's po...

In [8]:
lyrics.tail()

song  year      artist    genre  \
index                                                            
362232    who-am-i-drinking-tonight  2012  edens-edge  Country   
362233                         liar  2012  edens-edge  Country   
362234                  last-supper  2012  edens-edge  Country   
362235  christ-alone-live-in-studio  2012  edens-edge  Country   
362236                         amen  2012  edens-edge  Country   

                                                   lyrics  
index                                                      
362232  I gotta say\nBoy, after only just a couple of ...  
362233  I helped you find her diamond ring\nYou made m...  
362234  Look at the couple in the corner booth\nLooks ...  
362235  When I fly off this mortal earth\nAnd I'm meas...  
362236  I heard from a friend of a friend of a friend ...

### A Sample Lyric

In [12]:
lyrics['lyrics'][0]

"Oh baby, how you doing?\nYou know I'm gonna cut right to the chase\nSome women were made but me, myself\nI like to think that I was created for a special purpose\nYou know, what's more special than you? You feel me\nIt's on baby, let's get lost\nYou don't need to call into work 'cause you're the boss\nFor real, want you to show me how you feel\nI consider myself lucky, that's a big deal\nWhy? Well, you got the key to my heart\nBut you ain't gonna need it, I'd rather you open up my body\nAnd show me secrets, you didn't know was inside\nNo need for me to lie\nIt's too big, it's too wide\nIt's too strong, it won't fit\nIt's too much, it's too tough\nHe talk like this 'cause he can back it up\nHe got a big ego, such a huge ego\nI love his big ego, it's too much\nHe walk like this 'cause he can back it up\nUsually I'm humble, right now I don't choose\nYou can leave with me or you could have the blues\nSome call it arrogant, I call it confident\nYou decide when you find on what I'm working 

### A Look At the "year" Column

In [57]:
year_dist = pd.DataFrame(lyrics.year.value_counts())
year_dist.columns = ['Count']
year_dist['Percentage'] = lyrics.year.value_counts(normalize=True)
year_dist

Count  Percentage
2006  76590    0.211436
2007  63677    0.175788
2016  37409    0.103272
2015  28766    0.079412
2014  28250    0.077988
2008  21497    0.059345
2013  17331    0.047844
2012  15789    0.043587
2010  12106    0.033420
2011  12067    0.033312
2009  12029    0.033208
2005   5719    0.015788
2004   3698    0.010209
2003   2187    0.006037
2002   2158    0.005957
2000   1722    0.004754
1999   1586    0.004378
2001   1570    0.004334
1990   1362    0.003760
1997   1304    0.003600
1998   1285    0.003547
1996   1236    0.003412
1995   1160    0.003202
1994    968    0.002672
1993    918    0.002534
1992    907    0.002504
1991    559    0.001543
1977    552    0.001524
1971    541    0.001493
1973    520    0.001436
1989    511    0.001411
1972    509    0.001405
1982    498    0.001375
1970    429    0.001184
1979    420    0.001159
1981    414    0.001143
1974    410    0.001132
1978    407    0.001124
1980    401    0.001107
1988    388    0.001071
1986    365    0.001008
1975    364    0.001005
1983    359    0.000991
1976    344    0.000950
1985    333    0.000919
1987    307    0.000848
1984    298    0.000823
2038     10    0.000028
112       4    0.000011
702       1    0.000003
1968      1    0.000003
67        1    0.000003

<div class="alert alert-warning">
Potential Problem with this Dataset: Large portion of songs come from 2006, 2007 and 2016
</div>

### Remove Invalid "year"

### Look At the "genre" Column

In [58]:
genre_dist = pd.DataFrame(lyrics.genre.value_counts())
genre_dist.columns = ['Count']
genre_dist['Percentage'] = lyrics.genre.value_counts(normalize=True)
genre_dist

Count  Percentage
Rock           131377    0.362682
Pop             49444    0.136496
Hip-Hop         33965    0.093765
Not Available   29814    0.082305
Metal           28408    0.078424
Other           23683    0.065380
Country         17286    0.047720
Jazz            17147    0.047336
Electronic      16205    0.044736
R&B              5935    0.016384
Indie            5732    0.015824
Folk             3241    0.008947

<div class="alert alert-warning">
Potentail Problem with this Dataset: About 8% (29814) songs doesn't have a genre (Not Available)
</div>

### Number of Unique Artist

In [69]:
artist_dist = pd.DataFrame(lyrics.artist.value_counts())
artist_dist.columns = ['Count']
artist_dist['Percentage'] = lyrics.artist.value_counts(normalize=True)
artist_dist

Count  Percentage
dolly-parton                         755    0.002084
american-idol                        700    0.001932
elton-john                           680    0.001877
b-b-king                             667    0.001841
chris-brown                          655    0.001808
eddy-arnold                          628    0.001734
barbra-streisand                     624    0.001723
ella-fitzgerald                      623    0.001720
bob-dylan                            614    0.001695
bee-gees                             599    0.001654
david-bowie                          599    0.001654
dean-martin                          593    0.001637
eminem                               589    0.001626
celine-dion                          551    0.001521
frank-zappa                          550    0.001518
bruce-springsteen                    534    0.001474
elvis-costello                       534    0.001474
eric-clapton                         522    0.001441
beach-boys                           520    0.001436
bill-anderson                        473    0.001306
chicago                              461    0.001273
frank-sinatra                        435    0.001201
beatles                              429    0.001184
chamillionaire                       428    0.001182
50-cent                              425    0.001173
chumbawamba                          423    0.001168
britney-spears                       422    0.001165
diana-ross                           420    0.001159
emmylou-harris                       419    0.001157
cedarmont-kids                       417    0.001151
...                                  ...         ...
debauchery                             1    0.000003
count-monaco                           1    0.000003
dustin-smith                           1    0.000003
desiree                                1    0.000003
grapevine                              1    0.000003
cristopher-cross                       1    0.000003
cutaway                                1    0.000003
edita-abdieski                         1    0.000003
etches                                 1    0.000003
adriana-figueroa                       1    0.000003
abdelfattah-grini                      1    0.000003
ceza                                   1    0.000003
alabaster-box                          1    0.000003
four-seasons-one-day                   1    0.000003
altiyan-childs                         1    0.000003
e-miyavi                               1    0.000003
dianna-corcoran                        1    0.000003
ben-tankard                            1    0.000003
electric-river                         1    0.000003
gate-21                                1    0.000003
cali-cvsh                              1    0.000003
culture-shock                          1    0.000003
eric-zayne                             1    0.000003
denis-larose                           1    0.000003
the-audio-guy                          1    0.000003
abandoned-mall-arcade                  1    0.000003
gemini-major                           1    0.000003
dasco                                  1    0.000003
evita-new-broadway-cast-recording      1    0.000003
calypso-rose                           1    0.000003

[18231 rows x 2 columns]

In [65]:
len(lyrics.artist.value_counts())

18231

### Checking Number of Null Values

In [14]:
lyrics.isnull().sum()

song          2
year          0
artist        0
genre         0
lyrics    95680
dtype: int64

### Remove Rows With Missing Values

### Plot Distribution of Year

### Look for Specific Artist

In [71]:
lyrics.loc[lyrics.artist=='taylor-swift',:]

Empty DataFrame
Columns: [song, year, artist, genre, lyrics]
Index: []

<div class="alert alert-warning">
Potentail Problem with this Dataset: Missing Some Popular popstars like Taylor Swift
</div>